[#504] Implement SignName compatability with price restrictions

casptyche opened the pull request:
This is a proposed implementation for #504 Allow price limits for different item IDs.

The solution changes the PriceRestrictionModule to try to find a configured price restriction for the Item's ItemUtil#getSignName result. If none is found, it falls back to the item material as per the original implementation.

I've tested this briefly locally and it seems to work a treat with Slimefun4 items whilst maintaining compatibility with Vanilla items.


<!-- Issuehunt content -->

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#504: Allow price limits for different item ids such as Player Head#1 from Slimefun](https://oss.issuehunt.io/repos/1751509/issues/504)
- [#504: Allow price limits for different item ids such as Player Head#1 from Slimefun](https://oss.issuehunt.io/repos/1751509/issues/504)
---
</details>
<!-- /Issuehunt content-->

casptyche commented:
Updated MR to reflect it's ready for review and briefly tested locally

Phoenix616 commented on pom.xml:
```diff
@@ -6,7 +6,7 @@
     <groupId>com.acrobot.chestshop</groupId>
     <artifactId>chestshop</artifactId>
-    <version>3.12.3-SNAPSHOT</version>
+    <version>3.12.4-SNAPSHOT</version>
```
Please don't adjust the version, that will only be done when 3.12.3 was actually released.

casptyche replied:
Changed - will let you mark as resolved on evaluation

Phoenix616 commented on src/main/java/com/Acrobot/ChestShop/Listeners/Modules/PriceRestrictionModule.java:
```diff
@@ -17,14 +15,27 @@
+import static com.Acrobot.ChestShop.Permission.NOLIMIT_MIN_SELL;
+import static com.Acrobot.ChestShop.Permission.NOLIMIT_MIN_SELL_ID;
+import com.Acrobot.ChestShop.Signs.ChestShopSign;
+import com.Acrobot.ChestShop.Utils.ItemUtil;
```
These seem like a lot of line changes which I'm unsure they are all required. I assume your IDE did that automatically but I would prefer it if you'd only included changed imports that are actually required by the implemented function to keep the git history consistent. (I realise that the plugin does not have one consistent code style, adjusting that is on my todo but should be done globally imo)

casptyche replied:
Changed - will let you mark as resolved on evaluation

Phoenix616 commented on src/main/java/com/Acrobot/ChestShop/Listeners/Modules/PriceRestrictionModule.java:
```diff
@@ -73,12 +92,16 @@ private void load() {
+                ChestShop.getBukkitLogger().log(Level.WARNING,
+                        "If you want to automatically convert them you have to run this version on a pre 1.13 server.");
+                ChestShop.getBukkitLogger().log(Level.WARNING,
+                        "If you want to manually convert it and hide this message set the uses_materials key to true.");
```
Please keep the logs as they were before. I assume they conflicted with your IDE's max-line-length but that 
change makes it hard to see what actually changed in this commit so shouldn't be part of it.

casptyche replied:
Changed - will let you mark as resolved on evaluation

Phoenix616 commented on src/main/java/com/Acrobot/ChestShop/Listeners/Modules/PriceRestrictionModule.java:
```diff
@@ -89,7 +112,9 @@ private void convertToMaterial(String sectionPath) {
             for (String typeId : section.getKeys(false)) {
                 Material material = Material.matchMaterial(typeId);
                 if (material != null) {
-                    configuration.set(sectionPath + "." + material.toString().toLowerCase(Locale.ROOT), configuration.get(sectionPath + "." + typeId));
+                    configuration.set(
+                            sectionPath + "." + material.toString().toLowerCase(Locale.ROOT),
+                            configuration.get(sectionPath + "." + typeId));
```
Same line changes here that don't seem to change functionality.

casptyche replied:
Changed - will let you mark as resolved on evaluation

Phoenix616 reviewed (changes requested):
Thanks for your PR. From the concept it looks like it should be correct and work, just please adjust the comments (especially the version) and the formatting. (Please try to remove any line-change which only adjusted formatting not functionality. As a tip in general: Only including changed functions in a PR makes it way easier to review the functional changes ;))

casptyche commented:
My fault! I had removed the auto-format changes but it clearly auto-formatted again without me realizing right before I staged :')

Reverted import, version change & formatting changes.

Tested locally briefly to confirm still works (i.e. `Player Head#2` in priceLimits.yml is enforced in game, and `oak_sign` in priceLimits.yml is enforced in game, confirming that 'other items' and 'materials' both still work)

casptyche commented:
Is there any further action required on my end?

Phoenix616 commented:
Sorry, I simply didn't have time to look at this again yet.
